pageserver/src/tenant/timeline.rs

```diff
@@ -1502,29 +1508,9 @@ impl Timeline {
                 InMemoryLayerInfo::Open { .. } => {
                     // Upgrade to a write lock and freeze the layer
                     drop(layers_guard);
-                    let mut layers_guard = self.layers.write().await;
-                    let res = layers_guard.open_mut();
-
-                    let res = match res {
-                        Ok(open) => Ok(open
-                            .try_freeze_in_memory_layer(
-                                current_lsn,
-                                &self.last_freeze_at,
-                                &mut write_guard,
-                            )
-                            .await),
-                        Err(e) => Err(e),
-                    };
-
-                    // it is possible the LayerManager shutdown happened between us releasing the
-                    // read lock and re-acquiring the write lock; in such case, silently do
-                    // nothing.
-
-                    let res = match res {
-                        Ok(true) => self.flush_frozen_layers(current_lsn).map(|_req| ()),
-                        Ok(false) => Ok(()),
-                        Err(e) => Err(e.into()),
-                    };
```

Comment:
So all of this was: `freeze and if frozen, then signal`, which is now in `freeze_inmem_layer_at`.